### DISCUSSION TITLE
Silver ion sterilizer consumable in Dreame L10s Ultra

### DISCUSSION THREAD

**Discussion by ostrya** [Hypfer/Valetudo#1942] [Q&A / Support] [answered]
Hi, since a few days, my robot complains regularly that my Silver Ion sterilizer is consumed and needs to be replaced. I found out that in fact Dreame offers these sterilizers to add to the fresh water tank. But I don't have any installed and also don't plan to. But it would be nice if I was able to reset the consumable, because the robot sometimes randomly complains about the need to replace the sterilizer in the middle of the night.

I've found that there is something which sounds related in https://github.com/Hypfer/Valetudo/blob/25eaa76499d990f6d9b40898443c618117903b6e/backend/lib/robots/dreame/DreameMiotServices.js#L519, but it seems like that is not yet processed in Valetudo. 

On my device, I found /mnt/misc/consumable.json, which among others contains
```json
{"silver_degerm_time":1702042782,"silver_degerm_expire_time":1700596250,"silver_degerm_expired":1}
```

Would be incredible if you could add support for this consumable :-)

**Comment by Hypfer** [comment]
I'm a bit confused about this, because
- you're the first person to report it
- It did not happen on my robot (which I have since more than a year now)
- iirc the lifespan of that consumable is 365 days which is longer than the timespan the root is even available
- It is an optional consumable, meaning that it can't be on-by-default, because that would confuse users

So.. uh? Weird.
Did you by chance use the robot with the DreameHome app before?
Maybe you did enable something there?

I wouldn't want to add controls for that to the UI, because as you also said: `I don't have any installed and also don't plan to`.
I assume that no one will buy that consumable which would make that a dead UI element.

Still, weird that it happens on your machine and so far only there. For now, I'd suggest just editing that .json so that it's not expired.

**Reply by ostrya** [reply]
I got my robot on 2022-11-18. I did run it for a few days with the DreameHome app before rooting the device in December, but I definitely did not enable anything like that in the app.

I'll try out if modifying the json helps and post my results here.

**Reply by lindsaymarkward** [reply]
How did you go? :)

**Reply by ostrya** [reply, marked as answer]
Ah sorry. Yes, editing the JSON helped. After restarting the robot, the warning was gone. To be precise, I set the "silver_degerm_expired" value to 0 and added a trailing 0 to the "silver_degerm_expire_time" value to avoid it just expiring again. Not sure if just editing one of it would have been enough.
